app/src/org/commcare/android/nsd/MicroNode.java

```diff
@@ -0,0 +1,54 @@
+package org.commcare.android.nsd;
+
+import org.commcare.android.net.HttpRequestGenerator;
+import org.javarosa.core.io.StreamsUtil;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+
+/**
+ * API wrapper class for services provided by CommCare micronodes.
+ *
+ * Created by ctsims on 2/19/2016.
+ */
+public class MicroNode {
+    private final String mRoot;
+
+    private ArrayList<String[]> availableApplications;
+
+    public MicroNode(String uri) {
+        mRoot = uri;
+    }
+
+    public ArrayList<String[]> getAvailableApplications() {
```

Comment:
It would be nice if this returned an `ArrayList<Pair<String, String>>`, to help the reader know how many elements are in each array list entry